components/hal/cam_hal.c

```diff
@@ -0,0 +1,120 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "hal/cam_ll.h"
+#include "hal/cam_hal.h"
+#include "soc/soc_caps.h"
+
+/**
+ * @brief Configure line number to trigger interrupt
+ *
+ * @param hal CAM object data pointer
+ * @param num line number
+ *
+ * @return None
+ */
+static void cam_hal_set_line_int_num(cam_hal_context_t *hal, uint32_t num)
+{
+    if (num > 0) {
+        cam_ll_enable_hs_line_int(hal->hw, 1);
+        cam_ll_set_line_int_num(hal->hw, num);
+    } else {
+        cam_ll_enable_hs_line_int(hal->hw, 0);
+        cam_ll_set_line_int_num(hal->hw, 0);
+    }
+}
+
+/**
+ * @brief Configure V-SYNC filter threshold
+ *
+ * @param hal CAM object data pointer
+ * @param num V-SYNC filter threshold
+ *
+ * @return None
+ */
+static void cam_hal_set_vsync_filter_num(cam_hal_context_t *hal, uint32_t num)
+{
+    if (num > 0) {
+        cam_ll_enable_vsync_filter(hal->hw, 1);
+        cam_ll_set_vsync_filter_thres(hal->hw, num);
+    } else {
+        cam_ll_enable_vsync_filter(hal->hw, 0);
+        cam_ll_set_vsync_filter_thres(hal->hw, 0);
+    }
+}
+
+/**
+ * @brief Initialize CAM hardware
+ *
+ * @param hal    CAM object data pointer
+ * @param config CAM configuration
+ *
+ * @return None
+ */
+void cam_hal_init(cam_hal_context_t *hal, const cam_hal_config_t *config)
+{
+    memset(hal, 0, sizeof(cam_hal_context_t));
+
+    hal->hw = CAM_LL_GET_HW(0);
+
+    cam_ll_enable_stop_signal(hal->hw, 0);
+    cam_ll_swap_dma_data_byte_order(hal->hw, config->byte_swap_en);
+    cam_ll_reverse_dma_data_bit_order(hal->hw, 0);
+    cam_ll_enable_vsync_generate_eof(hal->hw, 1);
+
+    cam_hal_set_line_int_num(hal, 0);
+    cam_hal_set_vsync_filter_num(hal, 0);
+
+    cam_ll_enable_invert_pclk(hal->hw, 0);
+    cam_ll_set_input_data_width(hal->hw, 8);
+    cam_ll_enable_invert_de(hal->hw, 0);
+    cam_ll_enable_invert_vsync(hal->hw, 0);
+    cam_ll_enable_invert_hsync(hal->hw, 0);
+    cam_ll_set_vh_de_mode(hal->hw, 0); // Disable vh_de mode default
+    cam_ll_enable_rgb_yuv_convert(hal->hw, 0); // bypass conv module default
```

Comment:
I totally understand what you are saying. The HAL file was taken from ESP-IDF v5.3. For reasons like maintenance and compliance is better if we try to stick with the original code. In Zephyr, nothing stops us from take this values from DTS and call the LL API to proper configure the registers.

Reply:
So you suggest that we ignore the `cam_hal_init` function and implement our own version in the Zephyr video driver? 
This would involve including `<hal/cam_ll.h>` and manually initializing the  `cam_hal_context_t` struct. 
I don't see an issue with this approach. I just want to ensure we're on the same page.